integration/spark-interfaces/src/main/java/io/openlineage/spark/extension/v1/OlExprId.java

```diff
@@ -0,0 +1,39 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+package io.openlineage.spark.extension.v1;
+
+import java.util.Objects;
+
+/**
+ * Class to contain reference to Spark's ExprId without adding dependency to Spark library
+ *
+ * @see <a
+ *     href="https://github.com/apache/spark/blob/ce5ddad990373636e94071e7cef2f31021add07b/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala#L48">namedExpressions.scala</a>
+ */
+public final class OlExprId {
```

Comment:
Just thinking loudly. If we were to do that again I would start with Java version first. 
However, scala version is still marked as experimental. 

What if we made Scala interfaces depend on Java package and use Java classes instead of Scala case classes in both packages? After this change we would leave Scala interfaces `ColumnLevelLineageNode`, `InputLineageNode` and `OutputLineageNode` but their return type would be Java classes. 

I think it's the last moment to have it doable. This will simplify on the openlineage-spark connector side and reduce ricks of interface packages grow inconsistently. 

@mobuchowski does it make sense?

Reply:
I agree - we don't have any known implementation yet. Better to do it right then way overly cautious.